metadata/metadata_service/proxy/atlas_proxy.py

```diff
@@ -1674,4 +1674,4 @@ def get_popular_resources(self, *,
                               num_entries: int,
                               resource_types: List[str],
                               user_id: Optional[str] = None) -> Dict[str, List]:
-        pass
+        raise NotImplementedError
```

Comment:
If we want to base the logic on this exception then we need to:
1. remove `@abstractmethod` decorator
2. remove dummy implementation of the method from any proxy that doesn't support this (like AtlasProxy)

Otherwise:
1. Class won't be instantiated if method decorated with `@abstractmethod` is not overriden (`TypeError: Can't instantiate abstract class Actual with abstract methods` will be raised) 

Example:
```
import abc

class Test(abc.ABC):
    @abc.abstractmethod
    def x(self):
        raise NotImplementedError

class Actual(Test):
    def y(self):
        pass

x = Actual()
```

Reply:
@mgorsk1  Sure, this is another way of implementing things, but the way I implemented is not breaking anything, and I don't think this is wrong in any way. 

I want to make sure that each proxy has the method visible so as not to miss, and visible all the times that this needs to be implemented. 

```python
import abc

class Test(abc.ABC):
    @abc.abstractmethod
    def x(self):
        raise NotImplementedError

class Actual(Test):
    def y(self):
        pass

    def x(self):
        raise NotImplementedError

o = Actual()
```

Reply:
Sure, that makes sense as long as this PR also implements this method as not an abstract one for each of the proxies. Otherwise it's not really fixing anything in terms of backwards compatibility.

Reply:
Sorry, I don't get it what you said. We already override this method in each proxy.